source/commands/danger-run.ts

```diff
@@ -26,45 +26,54 @@ if (process.env["DANGER_VERBOSE"] || (program as any).verbose) {
   global.verbose = true
 }
 
-const source = getCISource(process.env, (program as any).externalCiProvider || undefined)
+// a dirty wrapper to allow async functionality in the setup
+async function run(): Promise<any> {
+  const source = getCISource(process.env, (program as any).externalCiProvider || undefined)
 
-if (!source) {
-  console.log("Could not find a CI source for this run")
-  // Check for ENV["CI"] and wanr they might want a local command instead?
-  process.exitCode = 1
-}
-
-if (source && !source.isPR) {
-  console.log("Skipping due to not being a PR")
-}
-
-if (source && source.isPR) {
-  const platform = getPlatformForEnv(process.env, source)
-  if (!platform) {
-    console.log(`Could not find a source code hosting platform for ${source.name}`)
+  if (!source) {
+    console.log("Could not find a CI source for this run")
+    // Check for ENV["CI"] and wanr they might want a local command instead?
     process.exitCode = 1
   }
+  // run the sources setup function, if it exists
+  if (source && source.setup) {
+    await source.setup()
+  }
 
-  if (platform) {
-    const dangerFile = (program as any).dangerfile || "dangerfile.js"
+  if (source && !source.isPR) {
+    console.log("Skipping due to not being a PR")
+  }
 
-    console.log(`OK, looks good ${source.name} on ${platform.name}`)
+  if (source && source.isPR) {
+    const platform = getPlatformForEnv(process.env, source)
+    if (!platform) {
+      console.log(`Could not find a source code hosting platform for ${source.name}`)
+      process.exitCode = 1
+    }
+
+    if (platform) {
+      const dangerFile = (program as any).dangerfile || "dangerfile.js"
 
-    try {
-      const stat = fs.statSync(dangerFile)
+      console.log(`OK, looks good ${source.name} on ${platform.name}`)
 
-      if (!!stat && stat.isFile()) {
-        d(`executing dangerfile at ${dangerFile}`)
-        const exec = new Executor(source, platform)
-        exec.setupAndRunDanger(dangerFile)
-      } else {
-        console.log(`looks like specified ${dangerFile} is not valid path`)
+      try {
+        const stat = fs.statSync(dangerFile)
+
+        if (!!stat && stat.isFile()) {
+          d(`executing dangerfile at ${dangerFile}`)
+          const exec = new Executor(source, platform)
+          exec.setupAndRunDanger(dangerFile)
+        } else {
+          console.log(`looks like specified ${dangerFile} is not valid path`)
+          process.exitCode = 1
+        }
+      } catch (error) {
         process.exitCode = 1
+        console.error(error.message)
+        console.error(error)
       }
-    } catch (error) {
-      process.exitCode = 1
-      console.error(error.message)
-      console.error(error)
     }
   }
 }
+
+run()
```

Comment:
Hah, I hadn't thought of that workaround, previously I had just avoided doing anything async in here